### DISCUSSION TITLE
Add support to inline-block custom componenet

### DISCUSSION THREAD

**Discussion by adamjosefus** [Thinkmill/keystatic#891] [Ideas]
Custom block are very handy. Every used component is wrapped in `<div>` element. That mean the specifiing display css property in custom component  have no sence.

I really wish/want add a inline custom component. For example for download file links or custom `<mark>` element.

**Comment by simonswiss** [comment]
Good news — a lot of new things are coming to Keystatic relatively soon, unlocked by a new (experimental for now) `Markdoc` field using a new editor.

It comes with a bunch of new "content components". That includes your wish for inline components and using custom `mark` tags 👍🏼 

https://docsmill.dev/npm/@keystatic/core@0.3.2#/content-components

![CleanShot 2024-01-18 at 20 00 00@2x](https://github.com/Thinkmill/keystatic/assets/485747/00539a9e-8075-45e1-b6f7-e890aa619d6a)

You can start using it today, but it's not documented yet, and still flagged as `__experimental`.

**Comment by ismaelrumzan** [comment]
@simonswiss Nice - I was able to set up a block in the admin editor. 
The content is currently saved as markdoc and I use the following to render the content of the content block. However, I'm not how to pass a component list to render the content components
```
            const { node } = await carditem.content();
            const renderable = Markdoc.transform(node, markdocConfig);
```
